src/components/Calculator.jsx

```diff
@@ -36,11 +38,54 @@ export const Calculator = () => {
         result: "Enter a valid number",
       });
 
-    return setData({ ...data, [v.target.name]: v.target.value });
+    const newData = { ...data, [v.target.name]: v.target.value };
+    const res = operations.operation(
+      parseFloat(newData.a),
+      parseFloat(newData.b),
+      newData.operation
+    );
+
+    const newHistory = [...history.slice(0, historyPos), newData];
+
+    if (newHistory.length >= 10) newHistory.shift();
+    newHistory.push({ ...newData, result: res });
+
+    setData({ ...newData, result: res });
+    setHistory(newHistory);
+    setHistoryPos(newHistory.length);
+  };
+
+  const handleSelect = (v) => {
+    const newData = { ...data, [v.target.name]: v.target.value };
+    const res = operations.operation(
+      parseFloat(newData.a),
+      parseFloat(newData.b),
+      newData.operation
+    );
+
+    setData({ ...newData, result: res });
+
+    const newHistoryPos = history.length;
+    setHistoryPos(newHistoryPos);
   };
 
-  const handleSelect = (v) =>
-    setData({ ...data, [v.target.name]: v.target.value });
+  React.useEffect(() => {
+    localStorage.setItem("calculatorHistory", JSON.stringify(history));
+    // eslint-disable-next-line
+  }, [history]);
+
+  React.useEffect(() => {
+    const storedHistory = JSON.parse(localStorage.getItem("calculatorHistory"));
+    if (storedHistory) setHistory(storedHistory);
+    // eslint-disable-next-line
+  }, []);
+
+  const operSimbols = {
```

Comment:
Change the variable name to OPERATION_SYMBOLS

Reply:
if it is already named correctly